core/src/test/scala/unit/kafka/admin/DescribeConsumerGroupTest.scala

```diff
@@ -315,24 +271,40 @@ class DescribeConsumerGroupTest extends KafkaServerTestHarness {
     val consumerGroupCommand = new KafkaConsumerGroupService(opts)
 
     TestUtils.waitUntilTrue(() => {
-        try {
           val (state, assignments) = consumerGroupCommand.describeGroup()
           state == Some("Stable") &&
           assignments.isDefined &&
           assignments.get.count(_.group == group) == 2 &&
           assignments.get.count{ x => x.group == group && x.partition.isDefined} == 2 &&
           assignments.get.count{ x => x.group == group && !x.partition.isDefined} == 0
-        } catch {
-          case _: GroupCoordinatorNotAvailableException | _: IllegalArgumentException =>
-            // Do nothing while the group initializes
-            false
-          case e: Throwable =>
-            throw e
-        }
       }, "Expected two rows (one row per consumer) in describe group results.")
 
     consumerGroupCommand.close()
   }
+
+  @Test
+  def testDescribeGroupWithNewConsumerWithShortInitializationTimeout() {
+    // run one consumer in the group consuming from a single-partition topic
+    val executor = new ConsumerGroupExecutor(brokerList, 1, group, topic)
+
+    // set the group initialization timeout too low for the group to stabilize
+    val cgcArgs = Array("--bootstrap-server", brokerList, "--describe", "--group", "group", "--timeout", "10")
+    val opts = new ConsumerGroupCommandOptions(cgcArgs)
+    val consumerGroupCommand = new KafkaConsumerGroupService(opts)
+
+    try {
+      val (state, assignments) = consumerGroupCommand.describeGroup()
+      fail("The consumer group command should fail due to low initialization timeout")
+    } catch {
+      case e @ (_: GroupCoordinatorNotAvailableException | _: IllegalArgumentException) =>
```

Comment:
There are two types of exceptions that can occur with a short timeout while the group initializes (like the one used in this unit test). One is `GroupCoordinatorNotAvailableException` for coordinator related errors ([here](https://github.com/apache/kafka/blob/913c09e4a94447a953cb55ccd702d7131d1764a8/core/src/main/scala/kafka/admin/AdminClient.scala#L72)). The other, `IllegalArgumentException`, occurs if the new group's protocol is not yet set ([here](https://github.com/apache/kafka/blob/913c09e4a94447a953cb55ccd702d7131d1764a8/core/src/main/scala/kafka/admin/AdminClient.scala#L177)). These two exceptions are the ones we are trying to circumvent here by retrying the corresponding API calls within the preset timeout. I hope it makes sense.